aten/src/ATen/TensorIndexing.h

```diff
@@ -230,22 +231,24 @@ static inline Tensor applySlice(
 static inline Tensor applySelect(
     const Tensor& self,
     int64_t dim,
-    int64_t index,
+    SymInt index,
     int64_t real_dim,
     const at::Device& /*self_device*/,
     const c10::optional<SymIntArrayRef>& self_sizes) {
+  auto opt_index_value = index.maybe_as_int();
   // See NOTE [nested tensor size for indexing]
-  if (self_sizes.has_value()) {
+  if (self_sizes.has_value() && opt_index_value.has_value()) {
+    int64_t index_value = opt_index_value.value();
     TORCH_CHECK_INDEX(
-        !(index == 0 && dim == 0 && self_sizes->empty()),
+        !(index_value == 0 && dim == 0 && self_sizes->empty()),
         "invalid index of a 0-dim tensor. ",
         "Use `tensor.item()` in Python or `tensor.item<T>()` in C++ to convert a 0-dim tensor to a number");
 
     auto size = (*self_sizes)[dim];
     TORCH_CHECK_INDEX(
-        size >= -index && size > index,
+        size >= -index_value && size > index_value,
```

Comment:
can't you just put some guards in here? Seems reasonable to do

Reply:
Or actually issuing it to `ShapeEnv` with `index < size`?

Reply:
Issue it to the shape env